src/core/thread.d

```diff
@@ -1716,7 +1716,12 @@ private:
         return cast(Mutex)_locks[1].ptr;
     }
 
-    __gshared align(Mutex.alignof) void[__traits(classInstanceSize, Mutex)][2] _locks;
+    @property static Mutex lowLevelThreadslock() nothrow @nogc
+    {
+        return cast(Mutex)_locks[2].ptr;
+    }
+
+    __gshared align(Mutex.alignof) void[__traits(classInstanceSize, Mutex)][3] _locks;
```

Comment:
Can this be its own private field to not interfere with #2411 (accessing this array causes crash with SPARC, SystemZ, PA-RISC, or any Solaris platforms).

Reply:
From what I can tell from other comments, you will also avoid being blocked by weka too.

Reply:
Thanks for the hint, now using a separate lock.